src/category_theory/monoidal/free/basic.lean

```diff
@@ -0,0 +1,365 @@
+/-
+Copyright (c) 2021 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import category_theory.monoidal.category
+import category_theory.groupoid
+
+universes v u
+
+namespace category_theory
+
+variables {C : Type u}
+
+section
+variables (C)
+
+inductive free_monoidal_category : Type u
+| of : C → free_monoidal_category
+| unit : free_monoidal_category
+| tensor : free_monoidal_category → free_monoidal_category → free_monoidal_category
+
+end
+
+notation `F` := free_monoidal_category
+
+inductive free_monoidal_category_hom : F C → F C → Type u
```

Comment:
Could we parametrise this by e.g. a quiver of generating morphisms? For application to the coherence theorem, we'll just take the empty generating set for morphisms, but it seems nice to have the general case. I think it just requires adding a single additional constructor.

Reply:
Maybe use namespaces here so you can just write `hom` and `hom_equiv`.

Reply:
> Could we parametrise this by e.g. a quiver of generating morphisms? For application to the coherence theorem, we'll just take the empty generating set for morphisms, but it seems nice to have the general case. I think it just requires adding a single additional constructor.
> 
> (edit: maybe this causes too much overhead later, having to repeatedly say you're talking about the non-generating-morphisms situation)

I don't think that the overhead would be particularly bad, but right now I'm finding it a bit awkward to state this. You can't just ask for a quiver instance on `C`, because if `C` is a category then it would do the wrong thing. You'd want to somehow pass in the quiver as an explicit argument. Do you have an application in mind that I could use as a test for the ergonomics?

Reply:
I think I retract my desire for a free monoidal category. I've definitely used free pivotal categories IRL, but that is a different kettle of fish (embedded planar graphs) than this anyway.

Reply:
We can always return to this later if the need arises.